Remotion/ObjectBinding/Core/BusinessObjectProvider.cs

```diff
@@ -151,38 +152,40 @@ protected BusinessObjectProvider (IBusinessObjectServiceFactory serviceFactory)
     ///    If your object model does not support services, this property should return an instance of type <see cref="NullDataStore{TKey,TValue}"/>.
     ///   </note>
     /// </remarks>
-    protected abstract IDataStore<Type, IBusinessObjectService> ServiceStore { get; }
+    protected abstract IDataStore<Type, IBusinessObjectService?> ServiceStore { get; }
 
     /// <summary>Gets the <see cref="IBusinessObjectServiceFactory"/> passed during construction.</summary>
     public IBusinessObjectServiceFactory ServiceFactory
     {
       get { return _serviceFactory; }
     }
 
-    public BusinessObjectProviderAttribute ProviderAttribute
+    public BusinessObjectProviderAttribute? ProviderAttribute
     {
       get { return _providerAttribute; }
     }
 
     /// <summary> Retrieves the requested <see cref="IBusinessObjectService"/>. Must not be <see langword="null" />.</summary>
-    public IBusinessObjectService GetService (Type serviceType)
+    public IBusinessObjectService? GetService (Type serviceType)
     {
       ArgumentUtility.CheckNotNullAndTypeIsAssignableFrom ("serviceType", serviceType, typeof (IBusinessObjectService));
 
-      IDataStore<Type, IBusinessObjectService> serviceStore = ServiceStore;
+      IDataStore<Type, IBusinessObjectService?> serviceStore = ServiceStore;
       Assertion.IsNotNull (serviceStore, "The ServiceStore evaluated and returned null. It should return a null object instead.");
 
       // Optimized for memory allocations
       if (_getServiceStoreValueFactory == null)
+      {
         _getServiceStoreValueFactory = type => _serviceFactory.CreateService (this, type);
+      }
 
       return serviceStore.GetOrCreateValue (serviceType, _getServiceStoreValueFactory);
     }
 
     /// <summary> Retrieves the requested <see cref="IBusinessObjectService"/>. </summary>
-    public T GetService<T> () where T: IBusinessObjectService
+    public T? GetService<T> () where T: IBusinessObjectService
     {
-      return (T) GetService (typeof (T));
+      return (T?) GetService (typeof (T));
```

Comment:
Re the nullability: I see you found the docs that explicitly allow null as a return value. nice.